src/Abstractions/NexusMods.Abstractions.Downloads/PersistedDownloadStatus.cs

```diff
@@ -0,0 +1,30 @@
+using JetBrains.Annotations;
+
+namespace NexusMods.Abstractions.Downloads;
+
+/// <summary>
+/// Status of a persisted download.
+/// </summary>
+[PublicAPI]
+public enum PersistedDownloadStatus : byte
+{
+    /// <summary>
+    /// Default status, the download is paused.
+    /// </summary>
+    Paused = 0,
+
+    /// <summary>
+    /// The download is progressing.
+    /// </summary>
+    Downloading = 1,
+
+    /// <summary>
+    /// The download is cancelled.
+    /// </summary>
+    Cancelled = 2,
+
+    /// <summary>
+    /// The download is completed.
+    /// </summary>
+    Completed = 3,
+}
```

Comment:
I don't think we are handling Download Failures well enough in the current system, and we are not doing anything to improve that in the new system.

What do you think would be the best way to handle a generic failure during download or during download start?
Should we just have a `PersistedDownloadStatus.Failed` state here?